City-of-Helsinki/helsinki-paatokset
# PP-289: Import motion data from meeting agenda items

This PR adds Drush commands for generating motions (decision nodes with incomplete data) from meeting agenda items.

**To test**
* Checkout branch
* Read code changes. Anything that doesn't make sense or that should be refactored?
* Run `make new` (take a database backup of your previous state if needed)
* SSH into the container and run `drush ap:fs;drush mim ahjo_meetings:latest;drush mim ahjo_cases:latest;drush mim ahjo_decisionmakers:all` (Note: do not import decisions yet!)
* Run `drush ap:gm --localdata -v`
  * This should find around 10 nodes and 105 items.
  * No items should fail, but around 20 should be skipped (agenda items without diary numbers)
  * Run the same command again. You should get 0 nodes because the meetings are marked as processed
* Check the decisions list: https://helsinki-paatokset.docker.so/fi/admin/content/decisions
  * The list should contain decisions without "content" data. The "is decision" field should also be false
  * Open some of these decisions. The layout should work normally, without errors
  * The color coding in the dropdown and org label should be correct
  * The "Appeal process" accordion should NOT appear in any of these nodes
  * The "other decisions for this meeting" link should work (as long as the meeting is relatively new, since we're testing with latest data)
    * For example: https://helsinki-paatokset.docker.so/fi/paattajat/kaupunginhallitus/asiakirjat/0040020221 and https://helsinki-paatokset.docker.so/fi/paattajat/tyoterveysliikelaitoksen-johtokunta/asiakirjat/U30110020221
* Note these specific motions:
  * https://helsinki-paatokset.docker.so/fi/asia/hel-2021-004219/c67aa8c7-3d20-c745-95b1-7e4384400002
  * https://helsinki-paatokset.docker.so/fi/asia/hel-2021-013148/C6DB4742-8C25-490E-9E1C-13CD60246CE6
  * https://helsinki-paatokset.docker.so/fi/asia/hel-2021-011266/1CACF5F5-B635-CCE8-89FC-7E43EC900002
* Run `drush mim ahjo_decisions:latest;drush ap:ud --localdata -v` (this will say that around 400 items failed, but this is fine since we're working with only the latest dataset)
  * Reload the previous motions (the URLs should redirect to new versions)
  * The native ID should be different now
  * The content data and appeal process accordion should now be visible. The motion content should now be moved to the accordions
* Go through the decisions list again: https://helsinki-paatokset.docker.so/fi/admin/content/decisions?title=&field_diary_number_value=&field_decision_native_id_value=&field_decision_motion_value=1&field_decision_content_value=All&order=field_diary_number&sort=desc
  * Some items are still motions (we're working with only latest data, and some motions belong to future meetings)
  * However, there shouldn't be any duplicates left (same diary number and meeting ID).
